web/src/store/modules/auth.ts

```diff
@@ -0,0 +1,71 @@
+import client from '@/services/client';
+
+export const Role = {
+  MODERATOR: 'moderator',
+  CONTRIBUTOR: 'contributor',
+  GUEST: 'guest',
+};
+
+export type RoleValue = 'moderator' | 'contributor' | 'guest';
+
+export interface AuthState {
+  user: any;
+  initialized: boolean;
+}
+
+const state: AuthState = {
+  user: null,
+  initialized: false,
+};
+
+const mutations = {
+  INITIALIZE(state: AuthState, { user }) {
+    state.user = user;
+    state.initialized = true;
+  },
+  LOGIN(state: AuthState, { user }) {
+    state.user = user;
+  },
+  LOGOUT(state: AuthState) {
+    state.user = null;
+  },
+};
+
+const actions = {
+  async login({ commit }, { email, password }) {
+    const response = await client.post('/v1/auth/login', { email, password });
+
+    commit('LOGIN', { user: response.data });
+  },
+  async logout({ commit }) {
+    commit('LOGOUT');
+    client.post('/v1/auth/logout');
+  },
```

Comment:
Should we not be committing `LOGOUT` after the API call itself has happened?
What if there are failures

Reply:
Yea I thought about this. I don't think it makes sense to wait. In most cases, logout will succeed. It's weird to see a loading state after clicking logout. This makes sense to me as is.

Reply:
hmm, I guess if logout doesn't happen for whatever reason, then when we check to see if user is logged in our not, it will say it is logged in, so i guess its okay